control-plane/csi-controller/src/main.rs

```diff
@@ -97,10 +97,21 @@ pub async fn main() -> Result<(), String> {
         CsiControllerConfig::get_config().rest_endpoint()
     );
 
-    server::CsiServer::run(
-        args.value_of("socket")
-            .expect("CSI socket must be specfied")
-            .to_string(),
-    )
-    .await
+    let mut signal_term = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::terminate())
+        .expect("Failed to register handler for SIGTERM");
+    let mut signal_int = tokio::signal::unix::signal(tokio::signal::unix::SignalKind::interrupt())
+        .expect("Failed to register handler for SIGINT");
+
+    tokio::select! {
+        _srv =  server::CsiServer::run(args.value_of("socket")
+            .expect("CSI socket must be specified")
+            .to_string()) => {
+            _srv?
```

Comment:
same here with the `_` probably not being needed

Reply:
I'm missing the reasoning behind this. This code does nothing, so why is it here? Also, why would we need to poll a future that does nothing, all the time? 

I guess it's related to coverage? If so, why is this not backed by a #cfg[(debug_assertions)]? And why dont we make the signal handler actually *do something*?

Reply:
fixed https://github.com/mayadata-io/mayastor-control-plane/pull/158/commits/2c281922104a3665655f0dddce06f56254e18e01